### PR TITLE
C99e99s61 understanding encapsulations polymorphisms in java

### PR DESCRIPTION
**Pull request by shubham9345**

### CODE REVIEW

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
+---
+title: "understanding the concept of Encapsulation and Polymorphism in java "
+date: '27 february, 2024'
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:3]: keep first letter of month in capital and change the format to month date, year (February 27, 2024)

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+sub_heading: "Let's dive into the concept of Encapsulation and Polymorphism more deeply with a lots of examples"
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:4]: update sub_heading with 'Encapsulation is a fundamental principle in object-oriented programming (OOP) where the internal state of an object is hidden from outside access.'. sub_heading should be the part of blog.

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+cover_image: ''
+category: 'general'
+---
+
+Encapsulation is a fundamental principle in object-oriented programming (OOP) where the internal state of an object is hidden from outside access. In Java, encapsulation is achieved using access modifiers.
+Let's dive into concept of Encapsulation and Polymorphism more deeply with a lots of examples.
+
+## What is Encapsulation ?
+
+Encapsulation is a process of binding the class of memeber into a single unit.It protects the data of class to be accessed from outside the class.However it is accessed outside the class by using access modifier let's see how.
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:14]: provide space before However

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+    // Setter for age
+    public void setAge(int age) {
+        if (age > 0) {
+            this.age = age;
+        } else {
+            System.out.println("Age cannot be negative.");
+        }
+    }
+     @Override
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:52]: provide space before override annotation

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    public String toString() {
+        return "Person{" +
+                "name='" + name + '\'' +
+                ", age=" + age +
+                '}';
+    }
+}
+// Main.java
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:60]: provide space before line 60

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+### Example:
+``` java
+// Parent class or Base class
+class Animal {
+    public void sound() {
+        System.out.println("Animal makes a sound");
+    }
+}
+// Subclass of Animal
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:101]: provide space after line 100

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+class Cow extends Animal {
+    @Override
+    public void sound() {
+        System.out.println("cow moo");
+    }
+}
+// subclass of Animal
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:108]: provide space after line 107

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+class Cat extends Animal {
+    @Override
+    public void sound() {
+        System.out.println("cat meows");
+    }
+}
+// Main.java
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:115]: give space after line 114

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+ Method with int parameter: 5.55
+ ```
+
+ ## Avantages of Polymorphism
+
+ 1. Increases code reusability by allowing objects of different classes to be treated as objects of a common class.
+
+ 2. Improves readability and maintainability of code by reducing the amount of code that needs to be written and maintained. 
+
+
```
Comment (pawan-malhotra) [inline general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:191]: update index count in readme.md file

**Review by pawan-malhotra — CHANGES_REQUESTED** [review]

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 1 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 2 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 3 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 4 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:53]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 5 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:62]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 6 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:104]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 7 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:112]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 8 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:120]: done

File: general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md (hunk 9 above)
Comment (shubham9345) [reply general/c99e99s61_understanding_encapsulation_and_polymorphism_in_java.md:196]: done

**Review by pawan-malhotra — APPROVED** [review]

**Pull request merged**